### DISCUSSION TITLE
Best way to do Entity Linking with heterogeneous data

### DISCUSSION THREAD

**Discussion by fabiofabris** [explosion/spaCy#7610] [Help: Best practices]
HI!

First, thanks a LOT for the awesome project.
I'm exploring the task of "entity linking" in the context of linking credit card transactions to merchants. The transactions contain a non-standardized merchant name and some meta-data, for instance: the transaction type (groceries, gas station, hotel, etc), the lat-long coordinates where the transition happened, and so on.

Assume that for some such transactions, I have the "ground truth" of which are the corresponding merchant's names and also some sort of meta-data, such as their location, their main business, and so on.

Given this scenario, what is the best way to link the transaction to a list of merchants, given that both (the transaction and the merchants) have non-text metadata?

In other words, if my document and my entities contain non-text metadata, what is a good way to integrate the metadata into an entity linking model?

**Comment by svlandeg** [comment]
Hi! I'm afraid there's currently no built-in mechanism in spaCy to account for generic meta data.

In general, you'd have to intervene at the step of instance creation / feature generation and concatenate the meta-data you think is useful, to the feature vectors before sending them off into your custom neural network. Currently spaCy's models are mainly focused on the content in the actual text though.
